ynh_systemd_action/ynh_systemd_action

```diff
@@ -28,32 +28,26 @@ ynh_systemd_action() {
     local log_path="${log_path:-/var/log/$service_name/$service_name.log}"
     local length=${length:-20}
     local timeout=${timeout:-300}
-    local wait_starting=true
 
-    if [[ -z "${line_match:-}" ]]
+    if [[ -n "${line_match:-}" ]]
     then
-        wait_starting=false
-    fi
-
-    ynh_clean_check_starting () {
-        # Stop the execution of tail.
-        kill -s 15 $pid_tail 2>&1
-        ynh_secure_remove "$templog" 2>&1
-    }
-
-    echo "Starting of $service_name" >&2
-    systemctl $action $service_name || ( journalctl --lines=$length -u $service_name >&2 && false)
-
-    if $wait_starting
-    then
-        # Following the starting of the app in its log
         local templog="$(mktemp)"
-        tail -F -n1 "$log_path" > "$templog" &
+	# Following the starting of the app in its log
+	if [ "$app_log" == "systemd" ] ; then
+            # Read the systemd journal
+            journalctl -u $service_name -f --since=-45 > "$templog" &
+	else
+            # Read the specified log file
+            tail -F -n0 "$log_path" > "$templog" &
+	fi
         # Get the PID of the tail command
         local pid_tail=$!
 
+	echo "${action^} the service $service_name" >&2
+	systemctl $action $service_name || ( journalctl --lines=$length -u $service_name >&2 && false)
```

Comment:
Maybe it could be good by example to show by example the last 10 line of `$log_path` if is it given in addtion to the systemd logs. And we could do the same thing in line 64.

Reply:
Why not, especially I encountered some issues with journalctl which was showing me another log.